### PR TITLE
fix(macos): menu bar

### LINKED ISSUE #6370: Menu bar on macOS after sign out -> sign in render bug
Seeing this intermittent menu bug on macOS that happens when:

- Launching for the first time
- Signing out and then back in

The bug disappears if I open and close the menu a few times.

Possibly it's a race condition? I.e. favorites are not instantiated before menu is created?

Screenshot 2024-08-20 at 8 43 58 AM

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #6370 

Tested with Jamil's patch. The bug is fixed on #6370 and no apparent regressions on iOS

### CODE REVIEW

File: swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift [hunk 1]
```diff
@@ -112,7 +112,7 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
     if case .authenticationCanceled = reason {
       do {
         // This was triggered from onDisconnect, so clear our token
-        Task { try await clearToken() }
```
Comment (ReactorScram) [inline swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift:115]: Unrelated to the menu, this was causing a compiler warning

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift [hunk 2]
```diff
@@ -115,7 +113,7 @@ struct SessionView: View {
       Text("Disconnecting...")
     case .reasserting:
       Text("No internet connection. Resources will be displayed when your internet connection resumes.")
-    case .invalid, .none:
```
Comment (ReactorScram) [inline swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift:118]: I guess `.none` === `nil`?

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift [hunk 3]
```diff
@@ -451,8 +452,14 @@ public final class MenuBar: NSObject, ObservableObject {
...
+      removeItemFromMenu(menu: menu, item: otherResourcesMenuItem)
+      removeItemFromMenu(menu: menu, item: otherResourcesSeparatorMenuItem)
+    } else {
+      let i = menu.index(of: aboutMenuItem)
+      addItemToMenu(menu: menu, item: otherResourcesMenuItem, at: i)
+      addItemToMenu(menu: menu, item: otherResourcesSeparatorMenuItem, at: i + 1)
+    }
 
     // Update the menu in place so everything won't vanish if it's open when it updates
     let diff = (newOthers).difference(
```
Comment (ReactorScram) [inline swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift:465]: I wonder how the indexes stay valid when we're iterating over the difference list

Comment (jamilbk) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift:465]: I had the same thought, maybe it's recalc'd on each iteration.

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift (hunk 2 above)
Comment (jamilbk) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift:118]: Yeah something like that.

**Review by jamilbk — APPROVED** [review]

**Pull request merged**